app/plugin.go

```diff
@@ -97,6 +97,11 @@ func (a *App) SyncPluginsActiveState() {
 				pluginEnabled = state.Enable
 			}
 
+			// Tie Apps proxy features to the feature flag.
+			if pluginId == "com.mattermost.apps" {
+				pluginEnabled = a.Config().FeatureFlags.AppsEnabled
+			}
```

Comment:
Does that mean an admin on Cloud can't disable the plugin? If so, is that obvious to an admin?

Reply:
We need to add some handling to this on the frontend. My approach would be to hide the plugin disable/enable button from the system console, or even the whole plugin.

Reply:
Then the questions becomes if admins should be able to remove the Apps plugin. @aaronrothschild What do you think?

Reply:
> Then the questions becomes if admins should be able to remove the Apps plugin. @aaronrothschild What do you think?

I have no objection on Cloud to this handling (ability to turn off the Plugin), but why does it need to be like that on Self-Managed?    I think Admins may expect to be able to turn off all their plugins if they want to.  We'd need to explain why the apps plugin, MUST remain running.

Reply:
@larkox Does it make sense to limit https://github.com/mattermost/mattermost-webapp/pull/7651 to cloud?

Reply:
I see value in that when we get out of the feature flag stage. Right now, the only way to keep the plugin running or stopped is by setting or unsetting the feature flag, doesn't matter if it is cloud or on-prem.